streambed-logged/src/lib.rs

```diff
@@ -605,6 +651,52 @@ fn find_offset(
     }
 }
 
+fn recover_active_file(
```

Comment:
Managed to create a comprehensive test where I write 3 records out, corrupt the 3rd one, restart the commit log to invoke recovery (verified via the debugger and by inspecting the file), and then producing a fourth and consuming the second and fourth.